tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -1455,4 +1455,67 @@ TEST(RDFSnapshotMore, ZeroOutputEntriesMT)
    gSystem->Unlink(fname);
 }
 
+// Test for custom basket size in Snapshot, ROOT - #17418
+TEST(RDFSnapshotMore, CustomBasketSize)
+{
+    // Create a simple ROOT file with a TTree
+    auto input_file = TFile::Open("input_file.root", "RECREATE");
+    auto tree = new TTree("tree", "Test Tree");
```

Comment:
There is a `new` keyword without the corresponding `delete`. This is wrong and in general `new` should be avoided at all costs, prefer stack-based variables or constructs that automatically manage memory such as `std::unique_ptr`